internal/controller/drplacementcontrol.go

```diff
@@ -1601,21 +1630,38 @@ func updatePeers(
 		}
 
 		for policyPeerClassIdx := range policyPeerClasses {
-			if policyPeerClasses[policyPeerClassIdx].StorageClassName ==
+			if policyPeerClasses[policyPeerClassIdx].StorageClassName !=
 				*vrgFromView.Status.ProtectedPVCs[pvcIdx].StorageClassName {
-				if hasPeerClass(
-					vrgPeerClasses,
-					*vrgFromView.Status.ProtectedPVCs[pvcIdx].StorageClassName,
-					policyPeerClasses[policyPeerClassIdx].ClusterIDs,
-				) {
-					break
-				}
+				continue
+			}
 
-				peerClasses = append(
+			if hasPeerClass(
+				vrgPeerClasses,
+				*vrgFromView.Status.ProtectedPVCs[pvcIdx].StorageClassName,
+				policyPeerClasses[policyPeerClassIdx].ClusterIDs,
+			) {
+				updatePeerClass(
+					log,
 					peerClasses,
 					policyPeerClasses[policyPeerClassIdx],
+					*vrgFromView.Status.ProtectedPVCs[pvcIdx].StorageClassName,
 				)
+
+				break
+			}
+
+			if hasPeerClass(
+				peerClasses,
+				*vrgFromView.Status.ProtectedPVCs[pvcIdx].StorageClassName,
+				policyPeerClasses[policyPeerClassIdx].ClusterIDs,
+			) {
+				break
 			}
+
+			peerClasses = append(
```

Comment:
if we are returning only the peerClasses slice then why do we need to check vrgPeerClasses?

Reply:
Initially `peerClasses` is set to `vrgPeerClasses`, later only if a peerClass form Policy is present in vrgPeerClasses is it updated in the peerClasses. Further, if a peerClass needs to be added to peerClasses, we check to ensure it is not already present, and only then append it. Older code in this space appended a missing peerClass for a new PVCa that now need protection multiple times, as the additional check at line 1679 was not present.